ggml-cuda/fattn.cu

```diff
@@ -231,32 +241,44 @@ static __global__ void flash_attn_ext_f16(
     // Pad internal representation of KQ, KQV to reduce shared memory bank conflicts:
     constexpr int D_padded = D + 8;
     constexpr int kqs_padded = FATTN_KQ_STRIDE + 8;
+    constexpr int kqar = sizeof(KQ_acc_t)/sizeof(half);
 
     const int gqa_ratio = ne02 / ne12; // With grouped query attention there are > 1 Q matrices per K, V matrix.
-    const float * Q_f   = (const float *) (Q + nb02* blockIdx.y);
+    const float * Q_f   = (const float *) (Q + nb02* blockIdx.y              + nb01*ic0);
     const half  * K_h   = (const half  *) (K + nb12*(blockIdx.y / gqa_ratio));
     const half  * V_h   = (const half  *) (V + nb12*(blockIdx.y / gqa_ratio)); // K and V have same shape
```

Comment:
I just remembered, for calculating the offsets I am not using `blockIdx.z` and `nb03`/`nb13` because I didn't understand what the purpose was and in the test cases ne3 was always 1. Are they used for continuous batching? If so, what is the expected memory layout?

Reply:
I think dim 3 is not used in `llama.cpp` during inference (I think it's used only during training and in other `ggml` projects). But it would be nice to take it into account. The easiest thing to do is add tests in `test-backend-ops` for that.

But in any case, I don't think this is causing the garbage output with 70B that @phymbert reports